### DISCUSSION TITLE
customer page cshtml file

### DISCUSSION THREAD

**Discussion by zprofreelancer** [serenity-is/Serenity#5382] [Help]
for example the customer page, where is the cshtml customer with the "new customer" button?

**Comment by minhhungit** [comment]
It's CustomerIndex.cshtml, if you check it you will see this bellow code:

```javascript
new Serene.Northwind.CustomerGrid($('#GridDiv'), {}).init();
```
Almost things you see on customer page is built from `CustomerGrid.ts`

"create new customer" button is also a button of toolbarButton in grid.ts

**CustomerGrid.ts**

```javascript
@Serenity.Decorators.registerClass()
    @Serenity.Decorators.filterable()
    export class CustomerGrid extends Serenity.EntityGrid<CustomerRow, any> {
       // ....
       // ....

        constructor(container: JQuery) {
            super(container);
        }

        getButtons() {
            // "CREATE NEW" BUTTON IS HERE <==================
            var buttons = super.getButtons(); 

           // you can add more buttons like this: 
            buttons.push(....);

            return buttons;
        }

    }
```

**Reply by zprofreelancer** [reply]
i can see the export excel and pdf button code in CustomerGrid.ts. But where the "new customer" button?... where the text "new customer" is embeded in code...also where is the search control and where is country an city dropdown list and the representative input text controls?

**Reply by arunkumar12712** [reply]
![image](https://user-images.githubusercontent.com/15244940/105578527-f7e70980-5da6-11eb-9a82-adabfbb59c85.png)

**Reply by zprofreelancer** [reply]
this is the customerrow file......i'm talking about the customer page design...

**Reply by minhhungit** [reply]
You better should read document here firstly 
https://serenity.is/docs

Read and do tutorials

**Reply by minhhungit** [reply]
For short answer, Serenity created it for us, Serenity create the button, Serenity create grid , it creates pager, it also create new form/edit form... 

It also exposed function that you can overwrite, for example above code, the function is `getButtons`

Check `Serenity.EntityGrid` class, you will see it

**Reply by zprofreelancer** [reply]
i'm reading the tutorial but it doesn't mention in details the classes used....anyway, i already have checked entitygrid and noticed that the new customer button is automatically created when i export class customergrid....I need to change the new customer text on the button...
I wonder how i do this since the button is created automatically....

Thanks

**Reply by arunkumar12712** [reply]
![image](https://user-images.githubusercontent.com/15244940/105606831-89637500-5dc1-11eb-9c69-3b6be5e8b47b.png)

**Reply by zprofreelancer** [reply]
Thanks alot... It works....this is not mentioned in the tutorial....are there any resources for customizing customer page??.....Also i need to customize the country and city and representatives controls..how to change the representative input text to dropdown control........how to do that please.
